sdks/java/core/src/main/java/org/apache/beam/sdk/state/MapState.java

```diff
@@ -77,6 +84,17 @@
    */
   ReadableState<V> get(K key);
 
+  /**
+   * A deferred lookup.
+   *
+   * <p>A user is encouraged to call {@code get} for all relevant keys and call {@code readLater()}
+   * on the results.
+   *
+   * <p>When {@code read()} is called, a particular state implementation is encouraged to perform
+   * all pending reads in a single batch.
+   */
+  ReadableState<V> getOrDefault(K key, @Nullable V defaultValue);
```

Comment:
wdyt about replacing this with either `computeIfAbsent` or `getOrDefault(K key, Supplier<V> defaultValue)`?

Either would allow users to avoid allocating throw-away defaultValues for every `getOrDefault` operation.

Reply:
Adding computeIfAbsent is a good idea - just added it.

while computeIfAbsent is a generalization off putIfAbsent, it's not quite a generalization of getOrDefault, since getOrDefault doesn't modify the map.